src/topp/DatabaseSuitability.cpp

```diff
@@ -196,292 +189,69 @@ class DatabaseSuitability :
-      }
-      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
-
-      // skip if top hit is out ouf FDR
-      if (scoreHigherThanFDR_(top_hit, FDR, q_value_score)) continue;
-
-      // check if top hit is found in de novo protein
-      if (!isNovoHit_(top_hit)) // top hit is db hit
-      {
-        ++count_db;
-        continue;
-      }
-
-      // find the second target hit, skip all decoy or novo hits inbetween
-      const PeptideHit* second_hit = nullptr;
-      String target = "target";
-      for (UInt i = 1; i < hits.size(); ++i)
-        {
-        // check for FDR
-        if (scoreHigherThanFDR_(hits[i], FDR, q_value_score)) break;
-
-        if (target.find(String(hits[i].getMetaValue("target_decoy"), 0)) == 0) // also check for "target+decoy" value
-        {
-          // check if hit is novo hit
-          if (isNovoHit_(hits[i])) continue;
-          
-          second_hit = &hits[i];
-          break;
-        }
-      }
-      if (second_hit == nullptr) // no second target hit with given FDR found
-      {
-        ++count_novo;
-        continue;
-      }
-
-      // second hit is db hit
-      ++count_interest;
-
-      // check for re-ranking
-      if (no_re_rank)
-      {
-        ++count_novo;
-        continue;
-      }
-
-      // check for xcorr score
-      if (!top_hit.metaValueExists("MS:1002252") || !second_hit->metaValueExists("MS:1002252"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No cross correlation score found at peptide hit. Only Comet search engine is supported right now."));
-      }
-      
-      double top_xscore_mw = double(top_hit.getMetaValue("MS:1002252")) / top_hit.getSequence().getMonoWeight();
-      double second_xscore_mw = double(second_hit->getMetaValue("MS:1002252")) / second_hit->getSequence().getMonoWeight();
-      if (top_xscore_mw - second_xscore_mw <= cut_off)
-      {
-        ++count_db;
-        ++count_re_ranked;
-      }
-      else
+      if (pep_id.getScoreType() != "novorscore")
       {
-        ++count_novo;
+        OPENMS_LOG_ERROR << in_novo << " contains at least one identification without a novorscore! Make sure this file contains only deNovo sequences." << endl;
+        return INPUT_FILE_CORRUPT;
       }
+      if (pep_id.getHits().empty()) continue;
+      unique_novo.insert(pep_id.getHits()[0].getSequence());
     }
 
-    double suitability = double(count_db) / (count_db + count_novo); //db suitability
+    Ms2IdentificationRate q;
+    q.compute(novo_peps, exp, true);
 
-    // spectra quality
+    Ms2IdentificationRate::IdentificationRateData spectral_quality = q.getResults()[0];
 
-    Size count_novo_seq = 0;
-    set<AASequence> unique_novo;
+    DBSuitability s;
+    Param p;
+    p.setValue("no_re_rank", no_re_rank ? "true" : "false");
```

Comment:
having duplicated parameters (even though its just 3) is bad for maintenance.
You can define the parameters in the class and simply make new TOPPSubsection 'algorithm' as done in other tools which just takes the Param from the class.
To pass the parameters, simply extract them back (done in many other tools) -- see MSSimulator etc..

Reply:
Yes, thanks, this is really important.